### ISSUE TITLE
[no-unused-expressions] Still false positive with optional chaining in v2.7.0

### ISSUE THREAD

**Issue by yangmingshan** [typescript-eslint/typescript-eslint#1220]
**Repro**

```JSON
{
  "rules": {
    "no-unused-expressions": "error"
  }
}
```

```TS
function test(cb?: () => void): void {
  cb?.();
}
```

**Expected Result**
No error

**Actual Result**
```
error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.7.0` |
| `@typescript-eslint/parser`        | `2.7.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `13.1.0` |
| `yarn`                              | `1.19.1` |

**Comment by a-tarasyuk** [comment]
@yangmingshan Do you use `no-unused-expressions` or `@typescript-eslint/no-unused-expressions`?

cc @bradzacher

**Comment by yangmingshan** [comment]
@a-tarasyuk `no-unused-expressions` will report error, `@typescript-eslint/no-unused-expressions` is work fine.

**Comment by a-tarasyuk** [comment]
@yangmingshan  I think, the issue can be closed, `@typescript-eslint/no-unused-expressions` [handles](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/no-unused-expressions.ts#L26) optional chaining

cc @bradzacher

**Comment by yangmingshan** [comment]
@a-tarasyuk I'm not sure, but if you all agree, you can close this issue, I'm ok.

**Comment by Bogidon** [comment]
Will
```json
"no-unused-expressions": "off",
"@typescript-eslint/no-unused-expressions": "error",
```
be added to `@typescript-eslint/recommended`?

**Comment by bradzacher** [comment]
unsure.
probably not - `no-unused-expressions` isn't a recommended rule in eslint core.

**Comment by Bogidon** [comment]
Ah whoops, my bad. The rule was coming from airbnb. Happen to know a library that does to airbnb what `@typescript-eslint/eslint-recommended` does to... `eslint:recommended`?

**Comment by bradzacher** [comment]
https://www.npmjs.com/package/eslint-config-airbnb-typescript

All you need to do is
`{ "extends": ["airbnb-typescript"] }`
